[BUG] `ROLLING_TEST` branch 25.02 fails when built with static cuda runtime

I ran cudf tests branch 25.02 on a GB100 node in a CUDA 12.8 container and got this failures:
```
[----------] 2 tests from GroupedRollingRangeOrderByFloatingPointTest/0, where TypeParam = float
[ RUN      ] GroupedRollingRangeOrderByFloatingPointTest/0.BoundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByFloatingPointTest/0.BoundedRanges, where TypeParam = float (0 ms)
[ RUN      ] GroupedRollingRangeOrderByFloatingPointTest/0.UnboundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByFloatingPointTest/0.UnboundedRanges, where TypeParam = float (0 ms)
[----------] 2 tests from GroupedRollingRangeOrderByFloatingPointTest/0 (0 ms total)

[----------] 2 tests from GroupedRollingRangeOrderByFloatingPointTest/1, where TypeParam = double
[ RUN      ] GroupedRollingRangeOrderByFloatingPointTest/1.BoundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByFloatingPointTest/1.BoundedRanges, where TypeParam = double (0 ms)
[ RUN      ] GroupedRollingRangeOrderByFloatingPointTest/1.UnboundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByFloatingPointTest/1.UnboundedRanges, where TypeParam = double (0 ms)
[----------] 2 tests from GroupedRollingRangeOrderByFloatingPointTest/1 (0 ms total)

[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/0, where TypeParam = numeric::fixed_point<int, (numeric::Radix)10>
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/0.BoundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/0.BoundedRanges, where TypeParam = numeric::fixed_point<int, (numeric::Radix)10> (0 ms)
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/0.UnboundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/0.UnboundedRanges, where TypeParam = numeric::fixed_point<int, (numeric::Radix)10> (0 ms)
[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/0 (0 ms total)

[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/1, where TypeParam = numeric::fixed_point<long, (numeric::Radix)10>
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/1.BoundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/1.BoundedRanges, where TypeParam = numeric::fixed_point<long, (numeric::Radix)10> (0 ms)
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/1.UnboundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/1.UnboundedRanges, where TypeParam = numeric::fixed_point<long, (numeric::Radix)10> (0 ms)
[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/1 (0 ms total)

[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/2, where TypeParam = numeric::fixed_point<__int128, (numeric::Radix)10>
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/2.BoundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/2.BoundedRanges, where TypeParam = numeric::fixed_point<__int128, (numeric::Radix)10> (0 ms)
[ RUN      ] GroupedRollingRangeOrderByDecimalTypedTest/2.UnboundedRanges
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByDecimalTypedTest/2.UnboundedRanges, where TypeParam = numeric::fixed_point<__int128, (numeric::Radix)10> (0 ms)
[----------] 2 tests from GroupedRollingRangeOrderByDecimalTypedTest/2 (0 ms total)

[----------] 8 tests from GroupedRollingRangeOrderByStringTest
[ RUN      ] GroupedRollingRangeOrderByStringTest.Ascending_Partitioned_NoNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Ascending_Partitioned_NoNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Ascending_NoParts_NoNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Ascending_NoParts_NoNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Ascending_Partitioned_WithNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Ascending_Partitioned_WithNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Ascending_NoParts_WithNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidDevice: invalid device ordinal" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Ascending_NoParts_WithNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Descending_Partitioned_NoNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Descending_Partitioned_NoNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Descending_NoParts_NoNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Descending_NoParts_NoNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Descending_Partitioned_WithNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Descending_Partitioned_WithNulls (0 ms)
[ RUN      ] GroupedRollingRangeOrderByStringTest.Descending_NoParts_WithNulls
unknown file: Failure
C++ exception with description "parallel_for failed: cudaErrorInvalidDevice: invalid device ordinal" thrown in the test body.
[  FAILED  ] GroupedRollingRangeOrderByStringTest.Descending_NoParts_WithNulls (0 ms)
[----------] 8 tests from GroupedRollingRangeOrderByStringTest (0 ms total)

[----------] 3 tests from GroupedRollingErrorTest
[ RUN      ] GroupedRollingErrorTest.NegativeMinPeriods
[       OK ] GroupedRollingErrorTest.NegativeMinPeriods (0 ms)
[ RUN      ] GroupedRollingErrorTest.EmptyInput
[       OK ] GroupedRollingErrorTest.EmptyInput (0 ms)
[ RUN      ] GroupedRollingErrorTest.SumTimestampNotSupported
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:510: Failure
Expected: cudf::grouped_rolling_window( grouping_keys, input_D, 2, 2, 0, *cudf::make_sum_aggregation<cudf::rolling_aggregation>()) throws an exception of type cudf::logic_error.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:514: Failure
Expected: cudf::grouped_rolling_window( grouping_keys, input_s, 2, 2, 0, *cudf::make_sum_aggregation<cudf::rolling_aggregation>()) throws an exception of type cudf::logic_error.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:518: Failure
Expected: cudf::grouped_rolling_window( grouping_keys, input_ms, 2, 2, 0, *cudf::make_sum_aggregation<cudf::rolling_aggregation>()) throws an exception of type cudf::logic_error.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:522: Failure
Expected: cudf::grouped_rolling_window( grouping_keys, input_us, 2, 2, 0, *cudf::make_sum_aggregation<cudf::rolling_aggregation>()) throws an exception of type cudf::logic_error.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:526: Failure
Expected: cudf::grouped_rolling_window( grouping_keys, input_ns, 2, 2, 0, *cudf::make_sum_aggregation<cudf::rolling_aggregation>()) throws an exception of type cudf::logic_error.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
[  FAILED  ] GroupedRollingErrorTest.SumTimestampNotSupported (0 ms)
[----------] 3 tests from GroupedRollingErrorTest (0 ms total)

[----------] 4 tests from GroupedRollingTest/0, where TypeParam = signed char
[ RUN      ] GroupedRollingTest/0.SimplePartitionedStaticWindowsWithGroupKeys
/rapids/cudf/cpp/tests/rolling/grouped_rolling_test.cpp:131: Failure
Expected: output = cudf::grouped_rolling_window( keys, input, preceding_window, following_window, min_periods, op) doesn't throw an exception.
  Actual: it throws thrust::system::system_error with description "parallel_for failed: cudaErrorInvalidResourceHandle: invalid resource handle".
CMake Error at /rapids/cudf/cpp/build/rapids-cmake/run_gpu_test.cmake:34 (execute_process):
  execute_process failed command indexes:

    1: "Abnormal exit with child return code: Segmentation fault"
```

--- a/cpp/tests/rolling/offset_row_window_test.cpp
+++ b/cpp/tests/rolling/offset_row_window_test.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021-2024, NVIDIA CORPORATION.
+ * Copyright (c) 2021-2025, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/cpp/tests/rolling/offset_row_window_test.cpp
+++ b/cpp/tests/rolling/offset_row_window_test.cpp
@@ -43,7 +43,6 @@ auto constexpr null = int32_t{0};  // NULL representation for int32_t;
 auto no_nulls_list() { return nulls_at({}); }
 
 struct OffsetRowWindowTest : public cudf::test::BaseFixture {
-
   struct rolling_runner {
     cudf::window_bounds _preceding, _following;
     cudf::size_type _min_periods;
@@ -54,7 +53,9 @@ struct OffsetRowWindowTest : public cudf::test::BaseFixture {
     rolling_runner(cudf::window_bounds const& preceding,
                    cudf::window_bounds const& following,
                    cudf::size_type min_periods_ = 1)
-      : _preceding{preceding}, _following{following}, _min_periods{min_periods_},
+      : _preceding{preceding},
+        _following{following},
+        _min_periods{min_periods_},
         _keys{0, 0, 0, 0, 0, 0, 1, 1, 1, 1},
         _values{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
     {

--- a/cpp/tests/rolling/offset_row_window_test.cpp
+++ b/cpp/tests/rolling/offset_row_window_test.cpp
@@ -54,8 +54,8 @@ struct OffsetRowWindowTest : public cudf::test::BaseFixture {
     rolling_runner(cudf::window_bounds const& preceding,
                    cudf::window_bounds const& following,
                    cudf::size_type min_periods_ = 1)
-      : _preceding{preceding}, _following{following}, _min_periods{min_periods_}, 
-        _keys{0, 0, 0, 0, 0, 0, 1, 1, 1, 1}, 
+      : _preceding{preceding}, _following{following}, _min_periods{min_periods_},
+        _keys{0, 0, 0, 0, 0, 0, 1, 1, 1, 1},
         _values{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
     {
     }
@@ -95,7 +95,7 @@ TEST_F(OffsetRowWindowTest, OffsetRowWindow_Grouped_3_to_Minus_1)
 {
   auto const preceding = cudf::window_bounds::get(3);
   auto const following = cudf::window_bounds::get(-1);
-  auto run_rolling     = rolling_runner{preceding, following}; 
+  auto run_rolling     = rolling_runner{preceding, following};
   run_rolling.min_periods(1).grouped(true);
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*run_rolling(*AGG_COUNT_NON_NULL),

--- a/cpp/tests/rolling/offset_row_window_test.cpp
+++ b/cpp/tests/rolling/offset_row_window_test.cpp
@@ -43,18 +43,20 @@ auto constexpr null = int32_t{0};  // NULL representation for int32_t;
 auto no_nulls_list() { return nulls_at({}); }
 
 struct OffsetRowWindowTest : public cudf::test::BaseFixture {
-  static ints_column const _keys;    // {0, 0, 0, 0, 0, 0, 1, 1, 1, 1};
-  static ints_column const _values;  // {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
 
   struct rolling_runner {
     cudf::window_bounds _preceding, _following;
     cudf::size_type _min_periods;
     bool _grouped = true;
+    ints_column const _keys;    // {0, 0, 0, 0, 0, 0, 1, 1, 1, 1};
+    ints_column const _values;  // {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
 
     rolling_runner(cudf::window_bounds const& preceding,
                    cudf::window_bounds const& following,
                    cudf::size_type min_periods_ = 1)
-      : _preceding{preceding}, _following{following}, _min_periods{min_periods_}
+      : _preceding{preceding}, _following{following}, _min_periods{min_periods_}, 
+        _keys{0, 0, 0, 0, 0, 0, 1, 1, 1, 1}, 
+        _values{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
     {
     }
 
@@ -80,9 +82,6 @@ struct OffsetRowWindowTest : public cudf::test::BaseFixture {
   };
 };
 
-ints_column const OffsetRowWindowTest::_keys{0, 0, 0, 0, 0, 0, 1, 1, 1, 1};
-ints_column const OffsetRowWindowTest::_values{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
-
 auto const AGG_COUNT_NON_NULL =
   cudf::make_count_aggregation<cudf::rolling_aggregation>(cudf::null_policy::EXCLUDE);
 auto const AGG_COUNT_ALL =
@@ -96,7 +95,8 @@ TEST_F(OffsetRowWindowTest, OffsetRowWindow_Grouped_3_to_Minus_1)
 {
   auto const preceding = cudf::window_bounds::get(3);
   auto const following = cudf::window_bounds::get(-1);
-  auto run_rolling     = rolling_runner{preceding, following}.min_periods(1).grouped(true);
+  auto run_rolling     = rolling_runner{preceding, following}; 
+  run_rolling.min_periods(1).grouped(true);
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*run_rolling(*AGG_COUNT_NON_NULL),
                                  ints_column{{0, 1, 2, 2, 2, 2, 0, 1, 2, 2}, nulls_at({0, 6})});
@@ -136,7 +136,8 @@ TEST_F(OffsetRowWindowTest, OffsetRowWindow_Ungrouped_3_to_Minus_1)
 {
   auto const preceding = cudf::window_bounds::get(3);
   auto const following = cudf::window_bounds::get(-1);
-  auto run_rolling     = rolling_runner{preceding, following}.min_periods(1).grouped(false);
+  auto run_rolling     = rolling_runner{preceding, following};
+  run_rolling.min_periods(1).grouped(false);
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*run_rolling(*AGG_COUNT_NON_NULL),
                                  ints_column{{0, 1, 2, 2, 2, 2, 2, 2, 2, 2}, nulls_at({0})});
@@ -176,7 +177,8 @@ TEST_F(OffsetRowWindowTest, OffsetRowWindow_Grouped_0_to_2)
 {
   auto const preceding = cudf::window_bounds::get(0);
   auto const following = cudf::window_bounds::get(2);
-  auto run_rolling     = rolling_runner{preceding, following}.min_periods(1).grouped(true);
+  auto run_rolling     = rolling_runner{preceding, following};
+  run_rolling.min_periods(1).grouped(true);
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(
     *run_rolling(*AGG_COUNT_NON_NULL),
@@ -219,7 +221,8 @@ TEST_F(OffsetRowWindowTest, OffsetRowWindow_Ungrouped_0_to_2)
 {
   auto const preceding = cudf::window_bounds::get(0);
   auto const following = cudf::window_bounds::get(2);
-  auto run_rolling     = rolling_runner{preceding, following}.min_periods(1).grouped(false);
+  auto run_rolling     = rolling_runner{preceding, following};
+  run_rolling.min_periods(1).grouped(false);
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*run_rolling(*AGG_COUNT_NON_NULL),
                                  ints_column{{2, 2, 2, 2, 2, 2, 2, 2, 1, null}, nulls_at({9})});

